app/models/emoji.rb

```diff
@@ -28,6 +28,20 @@ def self.standard
     Discourse.cache.fetch(cache_key("standard_emojis")) { load_standard }
   end
 
+  def self.allowed
+    denied_emojis = denied
+    all.reject { |e| denied_emojis.include?(e.name) }
```

Comment:
```suggestion
    denied_emojis = denied
    
    if denied_emojis.present?
      all.reject { |e| denied_emojis.include?(e.name) }
    end
```

I believe `Emoji.all` is quite a big array so we can just avoid looping through it when there is no denied emojis. Also should we cache the results here to avoid every request paying the cost of looping through a 1.8K items array on every page load?